extensions/spark/kyuubi-spark-connector-tpch/src/test/scala/org/apache/kyuubi/spark/connector/tpch/TPCHQuerySuite.scala

```diff
@@ -0,0 +1,138 @@
+ * {{{
+ *   build/mvn clean install \
+ *     -Dmaven.plugin.scalatest.exclude.tags="" \
+ *     -Dtest=none -DwildcardSuites=org.apache.kyuubi.spark.connector.tpch.TPCHQuerySuite
+ * }}}
+ *
+ * To re-generate golden files for this suite:
+ * {{{
+ *   KYUUBI_UPDATE=1 build/mvn clean install \
+ *     -Dmaven.plugin.scalatest.exclude.tags="" \
+ *     -Dtest=none -DwildcardSuites=org.apache.kyuubi.spark.connector.tpch.TPCHQuerySuite
+ * }}}
+ */
+// scalastyle:on line.size.limit
+
+@TpchTest
+class TPCHQuerySuite extends KyuubiFunSuite {
+
+  private val regenerateGoldenFiles = sys.env.get("KYUUBI_UPDATE").contains("1")
+
+  val baseResourcePath: Path =
+    Paths.get("extensions", "spark", "kyuubi-spark-connector-tpch", "src", "main", "resources")
+
+  val queries: Set[String] = (1 to 22).map(i => s"q$i").toSet
+
+  val queryToSumHash: Map[String, Long] = Map(
+    "q1" -> -2130215201L,
+    "q2" -> 2296723790L,
+    "q3" -> -4118618460L,
+    "q4" -> 1856406098L,
+    "q5" -> 3717321142L,
+    "q6" -> 2062248569L,
+    "q7" -> -1955579146L,
+    "q8" -> 453239528L,
+    "q9" -> 10861514367L,
+    "q10" -> -4090660469L,
+    "q11" -> -21773379672L,
+    "q12" -> 2455990065L,
+    "q13" -> -1379884230L,
+    "q14" -> 47333415L,
+    "q15" -> -2021679095L,
+    "q16" -> 326972717L,
+    "q17" -> 42L,
+    "q18" -> 1596157524L,
+    "q19" -> -1061725826L,
+    "q20" -> -35851308L,
+    "q21" -> 1407638530L,
+    "q22" -> 2111900859L)
+
+  private def fileToString(file: Path): String = {
+    new String(Files.readAllBytes(file), StandardCharsets.UTF_8)
+  }
+
+  test("run query on tiny") {
+    val viewSuffix = "view";
+    val sparkConf = new SparkConf().setMaster("local[*]")
+      .set("spark.ui.enabled", "false")
+      .set("spark.sql.catalogImplementation", "in-memory")
+      .set("spark.sql.catalog.tpch", classOf[TPCHCatalog].getName)
+    withSparkSession(SparkSession.builder.config(sparkConf).getOrCreate()) { spark =>
+      spark.sql("USE tpch.tiny")
+      queries.map { queryName =>
+        val in = getClass.getClassLoader.getResourceAsStream(
+          s"tpch/sql/$queryName.sql")
+        val queryContent: String = Source.fromInputStream(in)(Codec.UTF8).mkString
+        in.close()
+        queryName -> queryContent
+      }.foreach { case (name, sql) =>
+        try {
+          val result = spark.sql(sql).collect()
+          val schema = spark.sql(sql).schema
+          val schemaDDL = schema.toDDL + "\n"
+          spark.createDataFrame(result.toList.asJava, schema).createTempView(s"$name$viewSuffix")
+          val sumHashResult =
+            spark.sql(s"select sum(hash(*)) from $name$viewSuffix").collect().head.get(0)
+
+          val goldenFile = Paths.get(
+            baseResourcePath.toFile.getAbsolutePath,
+            "tpch",
+            "schema",
+            s"${name.stripSuffix(".sql")}.output.schema")
+
+          if (regenerateGoldenFiles) {
+            Files.write(goldenFile, schemaDDL.getBytes)
```

Comment:
Maybe we can also write an Apache License header to the golden files.

Reply:
![image](https://user-images.githubusercontent.com/14961757/172586467-42b3aa03-b073-478f-95f5-d5210a0e5d21.png)
Some other schema files don't seem to have added headers either.

Reply:
I see, we may remove them eventually.